sqs/runtime/src/main/java/io/quarkus/amazon/sqs/runtime/SqsDevServicesBuildTimeConfig.java

```diff
@@ -0,0 +1,17 @@
+package io.quarkus.amazon.sqs.runtime;
+
+import java.util.Set;
+
+import io.quarkus.amazon.common.runtime.DevServicesBuildTimeConfig;
+import io.quarkus.runtime.annotations.ConfigGroup;
+import io.quarkus.runtime.annotations.ConfigItem;
+
+@ConfigGroup
+public class SqsDevServicesBuildTimeConfig extends DevServicesBuildTimeConfig {
+
+    /**
+     * The queues to create on startup.
+     */
+    @ConfigItem(defaultValue = "default")
+    public Set<String> queues;
```

Comment:
I'm not familiar with SQS but would be it be useful to be able to configure the created queues? In this case, maybe this should be a `Map`-based config with the name being the key and a config object?

I'm not sure it makes sense though, just asking.

Reply:
I wanted to keep it as simple as the S3 part... but we could support queue attributes and tags.
See [CreateQueueRequest.Builder](https://sdk.amazonaws.com/java/api/latest/software/amazon/awssdk/services/sqs/model/CreateQueueRequest.Builder.html)

Reply:
BTW my PR is mostly a copy/paste of S3 devservice config ;-)

Reply:
I see some attributes like the queue type that would change the behaviour of the queue and could impact testing.
In s3, it was quite limited. Also, before dev services, the s3 extension already have automatic bucket creation so it was just refactored as is to fit in dev services.

Reply:
I tried to allow queue attribute config in `SqsDevServicesBuildTimeConfig` with 
```java
@ConfigItem
public Map<String, Map<String, String>> queueAttributes;
```
But I had `WARN  [io.qua.config] (main) Unrecognized configuration key "quarkus.sqs.devservices.queueAttributes.queue1" was provided; it will be ignored; verify that the dependency extension for this configuration is set or that you did not make a typo` error during tests with the following `application.properties`:
```properties
quarkus.sqs.devservices.queueAttributes.queue1=FifoQueue=true;ContentBasedDeduplication=true
```

So for now, we can only create basic SQS queue on devservices startup :-/

Reply:
Use queue-attributes. (Dash for word separator)

Reply:
Same error using
```properties
quarkus.sqs.devservices.queue-attributes.queue1=FifoQueue=true;ContentBasedDeduplication=true
```

`WARN  [io.qua.config] (main) Unrecognized configuration key "quarkus.sqs.devservices.queue-attributes.queue1" was provided; it will be ignored; verify that the dependency extension for this configuration is set or that you did not make a typo`

Reply:
> ```ini
> ontentBasedDeduplication=true
> ```

I guess this syntax do not work with map of map. Try with

````properties
quarkus.sqs.devservices.queues=queue1.fifo
quarkus.sqs.devservices.queue-attributes."queue1.fifo".FifoQueue=true
quarkus.sqs.devservices.queue-attributes."queue1.fifo".ContentBasedDeduplication=true
````

I'm not a big fan of `Map<String, String>` maybe we can support a subset of attributes with a strong typed sqs queue config class. This is how client configuration is done. A sub class like

```java
@ConfigGroup
    public static class SqsQueueDevServicesBuildTimeConfig {

        /**
         *  Designates a queue as FIFO
         */
        @ConfigItem(defaultValue = "false")
        public boolean fifoQueue;
    }
```

Would give

````properties
quarkus.sqs.devservices.queues=queue1.fifo
quarkus.sqs.devservices.queue-attributes."queue1.fifo".fifo-queue=true

````

There is around 20 attribute names. @gsmet  WDYT ?

Reply:
I tried other way to configure queue's attributes but I still get exceptions :-(

See https://github.com/quarkiverse/quarkus-amazon-services/commit/4d1f7225cf6081adce33e1d8e2caddf2866da940#diff-5a0545d3859ca1195c0de0ba0868e65b4838936beb52eaa38b2dce2b085ee218

```
IllegalArgumentException: SRCFG00013: No Converter registered for class io.quarkus.amazon.sqs.runtime.SqsDevServicesBuildTimeConfig$QueueConfig
```

Reply:
Maybe as a first iteration we could merge this PR that allow simple queue creation on devservices startup and then we could work on another PR to configure these queue's attributes?

Reply:
Yes, the attributes map is on a different key so there won't be any change. Also, I'm thinking of having a codegen to produce config class from AWS sdk builder class. This would benefit other extensions.